src/middleware.coffee

```diff
@@ -1,4 +1,5 @@
 async = require 'async'
+Promise = global.Promise || require('es6-promise').Promise
```

Comment:
I think https://github.com/stefanpenner/es6-promise#auto-polyfill means you could require the auto-polyfill, and then just use `Promise`. On the other hand, using this style is a lot more clear on where Promise is coming from.